src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -35,7 +40,7 @@ internal static SafeFileHandle Open(string path, Interop.Sys.OpenFlags flags, in
             bool enoentDueToDirectory = (flags & Interop.Sys.OpenFlags.O_CREAT) != 0;
 
             // Open the file. 
-            int fd;
+            IntPtr fd;
```

Comment:
Hmm.  It shouldn't happen very often (basically when the process receives a signal during the invocation of one of these operations), but I agree if it was more prevalent we'd want to avoid those costs.  Why don't we stick with what you have for now, but then separately do some testing, instrumenting our checks for EINTR to track how often it happens, and then running some workloads to see how often it happens.  I like the cleaniness that your PR here introduces, and it'd be nice to take it as far as possible.